Codeinwp/optimole-wp
# Best format option

### All Submissions:

 

### Changes proposed in this Pull Request:
 - Adds **Best format** toggle under **Settings > Compression**;
 - This option should be enabled by default for both new and old users;
 - This PR's full functionality depends on [this PR](https://github.com/Codeinwp/optimole-lambda-functions/pull/460);
 - Removes the old `format` flag logic - which was used for requesting AVIF files previously;
 - Reworks filter that was used to exclude extensions from using AVIF;
 

Closes Codeinwp/optimole-service#983.

### How to test the changes in this Pull Request:

1. Check that the **Best format** option exists and is on by default for both old users that updated the plugin, and new ones;
2. When the option is enabled, all image URLs that get replaced should contain the `f:best` flag;
3. If the option is off, the flag should not be present in the URLs;
4. When CSS/JS optimization is enabled, those files should still have `f:css` and `f:js`, regardless of the best format setting;
5. SVG files (using the SVG enable plugin) should have the `ig:avif` flag automatically now;
6. GIF files that have at least 20px in width and height should be replaced with `<video>` tags, and the `f:mp4` format should be used when the **Enable Gif to Video conversion** option is enabled;

I have deployed this to a instaWP instance to test URLs work as expected: https://ed-opossum-bahi.instawp.xyz/best-format-test/

<details>

<summary>InstaWP details</summary>
URL :
https://ed-opossum-bahi.instawp.xyz

Username :
tatizidigu7602

Password :
Y4z1MAaKpWn7x6rgv5TQ
</details>

### Other information:
